### DISCUSSION TITLE
Uvicorn keeps restarting (aka #364 seems to be back)

### DISCUSSION THREAD

**Discussion by mikkokotila** [Kludex/uvicorn#1465] [Potential Issue]
I have two systems that are identical in every other way, as much as two systems can be, but the other is running `0.17.4` and other is running whatever gets installed right now (I can't check as it's not possible to get into the docker container during the restart loop caused by Uvicorn). This issue first appeared last week when I re-deployed the system where this issue manifests, so the Uvicorn version would be one which gets installed since then so supposedly `0.17.6` which also match the working hypothesis here, as prior to last week the system was running a version prior to March 11th, 2022 (on which `0.17.6` was released). 

To me it seems to be like #364 which then re-emerged in #852.

**Comment by mikkokotila** [comment]
This is really not a discussion, it is 1) a bug, and 2) a regression from previous fixes. I have now tested locking the version to `0.17.4` without changing anything else, and that addressed the problem. This proves that `0.17.6` (or `0.17.5`) introduced 1) this bug, and 2) a regression from previous fixes in respect to #364 and #852.

**Reply by Kludex** [reply]
![imageedit_3_6118495132](https://user-images.githubusercontent.com/7353520/166066648-700f2cab-d0fe-4df6-9589-cce4d33548bb.jpeg)

**Reply by mikkokotila** [reply]
Yes, I saw that. It said "usually", but in this case, it seemed like a clear cut bug/regression, and given regression is generally considered more serious than just a bug, and bug alone would constitute an issue, it seemed like the right way to start an issue to save everyone's time. If the intention is that everything under all circumstances starts as a discussion, better to remove the "usually" from the template. It is a good idea though, to triage things through discussions as for well maintained packages most things end up being discussions and not actual issues 👍

**Reply by mikkokotila** [reply]
Only now I realize there is also this: 

![Screenshot 2022-04-29 at 23 49 32](https://user-images.githubusercontent.com/7943188/166067295-98e8f970-c612-43c0-b2ab-c1d19adcdada.png)

I guess it's hard to miss as the eye is trained to just pay attention to the actual issue boilerplate.

**Comment by euri10** [comment]
I don't think there's anything between 0.17.4 and 0.17.6 we changed that might affect this,
do you have a minimal reproducible example and a clear cut description of the issue, is it on docker only, is it not etc

**Reply by mikkokotila** [reply]
It was indeed resolved simply by changing the version, so there is something that changes there. My guess is that resource use is somehow affected, which then causes resources to run out, and uvicorn to restart.

This is the application in question: https://github.com/Lotus-King-Research/Padma-Backend

I don't think there is another way to replicate it than running this on EC2 with 2gb of RAM, as that is the environment where the issue manifests.
